modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/mapping/MappingServiceImpl.java

```diff
@@ -147,7 +152,7 @@ private CompletableFuture<List<MappedFragment>> map0(MultiStepPlan multiStepPlan
             return val;
         });
 
-        return cacheValue.mappedFragments;
+        return cacheValue.mappedFragments.thenApply(mappedFragments -> applyPartitionPruning(mappedFragments, parameters));
```

Comment:
I think when a query has no dynamic parameters, when can apply partition pruning to before adding fragments to cache.
And if there are dynamic parameters, we should apply partition pruning after retrieving fragments from the cache.

Reply:
seems dynamic parameters need to be part of cache key ?